### PR TITLE
Move BigInt JSON parse and strigify functions to `rpc-spec-types`

### PR DESCRIPTION
**Pull request by lorisleiva**
This PR moves the `parseJsonWithBigInts` and `stringifyJsonWithBigints` functions from the `rpc-transport-http` package to the `rpc-spec-types` package in order to reuse them within RPC Subscriptions packages.

Note that the library now explicitly re-exports a subset of the `rpc-spec-types` to avoid flooding the namespace with internal type and functions.

### CODE REVIEW

**Comment by lorisleiva** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3443?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3443** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3443?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3435** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3435?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3442](https://github.com/solana-labs/solana-web3.js/pull/3442) <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3442?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lorisleiva and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/rpc-transport-http/tsconfig.json [hunk 1]
```diff
@@ -5,5 +5,9 @@
     },
     "display": "@solana/rpc-transport-http",
     "extends": "../tsconfig/base.json",
-    "include": ["src"]
+    "include": [
+        "src",
+        "../rpc-spec-types/src/parse-json-with-bigints.ts",
+        "../rpc-spec-types/src/stringify-json-with-bigints.ts"
```
Comment (steveluscher) [inline packages/rpc-transport-http/tsconfig.json:11]: Damn you, VS Code.

**Review by mcintyre94 — APPROVED** [review]

**Review by steveluscher — APPROVED** [review]
Added via Giphy

**Comment by lorisleiva** [comment]
### Merge activity

* **Oct 27, 6:03 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3443).

**Pull request merged**